sdk/core/ts-http-runtime/src/client/clientHelpers.ts

```diff
@@ -45,6 +45,7 @@ export function addCredentialPipelinePolicy(
     const tokenPolicy = bearerTokenAuthenticationPolicy({
       credential,
       scopes: clientOptions?.credentials?.scopes ?? `${endpoint}/.default`,
+      allowInsecureConnection: clientOptions?.allowInsecureConnection ?? false,
```

Comment:
Could be missing context here, but I am a tad concerned about plumbing through the option from the client options like this. This change will result in a behavioral difference between the Azure and unbranded Core packages that is difficult to reconcile -- when we get around to making the Azure packages depend on the unbranded package we will need to have a way to set `allowInsecureConnection` _without_ having this affect the `bearerTokenAuthenticationPolicy`, in order to preserve the existing behavior. I have two ideas to resolve this issue, not sure if you have a preference or any other thoughts here:

1. Add another property on the client options, `allowInsecureBearerTokenAuth` or similar, to allow this behavior, and this additional property gets plumbed through instead, so that the behavior of `allowInsecureConnection` is unchanged. Azure Core can just not expose the new property.
1. Don't plumb the property through; if folks want to set `allowInsecureConnection` on the `bearerTokenAuthPolicy` they can remove the policy and re-add their own. This is obviously more complicated UX but perhaps that's not the worst thing honestly given we don't really want people doing this unless they truly have to?

Reply:
Thanks Timo! You're quite right on all accounts here. I know the concern for (2) was that injecting the policy will not be desired for Chat Protocol, it will be a significant complication to sample code. 

(1) does sound enticing, @bterlson what do you think? Any preferences here since you're likely closest to the folks asking for this?

Reply:
2 is off the table. I'm not opposed to 1, but I would prefer to get clarity on whether we can't just have this for Azure's policy as well. I think we are assuming that we can't for reasons, but can we verify?

Reply:
Yes, my comment is assuming that we want to keep the status quo for Azure -- as I said, I am lacking context... The concern appears related to being able to intercept an active AAD token due to this ([1](https://github.com/Azure/azure-sdk-for-js/pull/25207#pullrequestreview-1335754093), [2](https://github.com/Azure/azure-sdk-for-js/pull/17517#issuecomment-918635850)). I am no security expert but the reasoning seems logical to me -- that's not to say we can't revisit it. @joheredi may have more context since he authored the original PR to enable this which was reverted?

Clearly I am missing a lot of background around the motivation for this feature and why we need it now -- it sounds like we have a customer looking to use the unbranded package and this is a requirement for them? Is there a way for me to learn more about this? Just so I can understand what problem we are trying to solve here.

Reply:
Quick update here, we engaged the azsdksecurity folks and got their 👍 on the changes. I think we're good to merge this now. @timovv I'll include you in the internal thread for context.

@xirzec would appreciate your review / input on this

Reply:
Are we making this change for Azure as well then?

Reply:
Good Q Timo, let me get more clarity on that and get back to you